src/Services/PreHashStrings/LegacyPreHashString.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace LearnositySdk\Services\PreHashStrings;
+
+use LearnositySdk\Exceptions\ValidationException;
+
+class LegacyPreHashString implements PreHashStringInterface
```

Comment:
I assume there will be another class implementing this in the future, otherwise the interface makes no sense

Reply:
Yep, the idea is to make each signature pattern independent, rather than the current pattern we have which is really hard to audit because it jumps around and has too many ifs.

The LegacyPreHashString implementation is different from what we had in the SDK before, as it takes the approach of the implementation we have in LP instead, which is a lot simpler to follow.